### PR TITLE
Add comment request can post a message or a tag message, but not both…

### PR DESCRIPTION
**Pull request by jdizoglio**
…. Added code to the request evaluate if a tag message exists

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @jdizoglio, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by jdizoglio** [comment]
@boxtcarpel

**Comment by jdizoglio** [comment]
@boxtcarpel

**Comment by RicoYao** [comment]
+2 is there a reasonable place we can add a comment for this. Perhaps the BOXCommentAddRequest header?

**Comment by jdizoglio** [comment]
"CLA signed"

**Comment by boxcla** [comment]
Verified that @jdizoglio has just signed the CLA. Thanks, and we look forward to your contribution.

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m [hunk 1]
```diff
@@ -35,9 +35,15 @@ - (BOXAPIOperation *)createOperation
...
+    NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
+    
+    if (self.requestAllItemFields) {
+        parameters[BOXAPIParameterKeyFields] = [self fullCommentFieldsParameterString];
+    }
+    
     BOXAPIJSONOperation *JSONOperation = [self JSONOperationWithURL:URL
                                                          HTTPMethod:BOXAPIHTTPMethodGET
-                                              queryStringParameters:nil
+                                              queryStringParameters:parameters
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m:46]: Could you add to BOXFileCommentsRequestTests to ensure the params are set.

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 @interface BOXFileCommentsRequest : BOXRequestWithSharedLinkHeader
 
+@property (nonatomic, readwrite, assign) BOOL requestAllItemFields;
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.h:10]: Should we add the same to BOXCommentRequest?

Comment (jdizoglio) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.h:10]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m (hunk 1 above)
Comment (jdizoglio) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m:46]: Added the unit test and also for the new BOXCommentRequest all fields.

**Comment by RicoYao** [comment]
LGTM. Thank you for adding the tests.

**Pull request merged**